conf/simulator/nps/nps_sensors_params_common.h

```diff
@@ -0,0 +1,322 @@
+#ifndef NPS_MAG_MIN
+#define NPS_MAG_MIN -65536
+#endif
+#ifndef NPS_MAG_MAX
+#define NPS_MAG_MAX  65536
+#endif
+
+#define NPS_MAG_IMU_TO_SENSOR_PHI    0.
+#define NPS_MAG_IMU_TO_SENSOR_THETA  0.
+#define NPS_MAG_IMU_TO_SENSOR_PSI    0.
+
+#define NPS_MAG_SENSITIVITY_NUM  1
+#define NPS_MAG_SENSITIVITY_DEN  1
+#define NPS_MAG_SENSITIVITY      ((float)NPS_MAG_SENSITIVITY_NUM / (float)NPS_MAG_SENSITIVITY_DEN)
+
+#ifdef IMU_MAG_X_SENS
+#define NPS_MAG_SENSITIVITY_XX   MAG_BFP_OF_REAL(1./IMU_MAG_X_SENS)
+#else
+#define NPS_MAG_SENSITIVITY_XX   MAG_BFP_OF_REAL(1./NPS_MAG_SENSITIVITY)
+#endif
+#ifdef IMU_MAG_Y_SENS
+#define NPS_MAG_SENSITIVITY_YY   MAG_BFP_OF_REAL(1./IMU_MAG_Y_SENS)
+#else
+#define NPS_MAG_SENSITIVITY_YY   MAG_BFP_OF_REAL(1./NPS_MAG_SENSITIVITY)
+#endif
+#ifdef IMU_MAG_Z_SENS
+#define NPS_MAG_SENSITIVITY_ZZ   MAG_BFP_OF_REAL(1./IMU_MAG_Z_SENS)
+#else
+#define NPS_MAG_SENSITIVITY_ZZ   MAG_BFP_OF_REAL(1./NPS_MAG_SENSITIVITY)
+#endif
+
+#ifdef IMU_MAG_X_NEUTRAL
+#define NPS_MAG_NEUTRAL_X  IMU_MAG_X_NEUTRAL
+#else
+#define NPS_MAG_NEUTRAL_X  0
+#endif
+#ifdef IMU_MAG_Y_NEUTRAL
+#define NPS_MAG_NEUTRAL_Y  IMU_MAG_Y_NEUTRAL
+#else
+#define NPS_MAG_NEUTRAL_Y  0
+#endif
+#ifdef IMU_MAG_Z_NEUTRAL
+#define NPS_MAG_NEUTRAL_Z  IMU_MAG_Z_NEUTRAL
+#else
+#define NPS_MAG_NEUTRAL_Z  0
+#endif
+
+#ifdef IMU_MAG_X_SIGN
+#define NPS_MAG_SIGN_X  IMU_MAG_X_SIGN
+#else
+#define NPS_MAG_SIGN_X  1
+#endif
+#ifdef IMU_MAG_Y_SIGN
+#define NPS_MAG_SIGN_Y  IMU_MAG_Y_SIGN
+#else
+#define NPS_MAG_SIGN_Y  1
+#endif
+#ifdef IMU_MAG_Z_SIGN
+#define NPS_MAG_SIGN_Z  IMU_MAG_Z_SIGN
+#else
+#define NPS_MAG_SIGN_Z  1
+#endif
+
+
+#ifndef NPS_MAG_DT
+#define NPS_MAG_DT (1./100.)
+#endif
+
+
+/*
+ *  Barometer (pressure and std dev in Pascal)
+ */
+#define NPS_BARO_DT              (1./50.)
+
+/*
+ *  GPS
+ */
+
+#ifndef GPS_PERFECT
+#define GPS_PERFECT 1
+#endif
+
+#if GPS_PERFECT
+
+#define NPS_GPS_SPEED_NOISE_STD_DEV            0.
+#define NPS_GPS_SPEED_LATENCY                  0.
+#define NPS_GPS_POS_NOISE_STD_DEV              0.001
+#define NPS_GPS_POS_BIAS_INITIAL_X             0.
+#define NPS_GPS_POS_BIAS_INITIAL_Y             0.
+#define NPS_GPS_POS_BIAS_INITIAL_Z             0.
+#define NPS_GPS_POS_BIAS_RANDOM_WALK_STD_DEV_X 0.
+#define NPS_GPS_POS_BIAS_RANDOM_WALK_STD_DEV_Y 0.
+#define NPS_GPS_POS_BIAS_RANDOM_WALK_STD_DEV_Z 0.
+#define NPS_GPS_POS_LATENCY                    0.
+
+#else
+
+#define NPS_GPS_SPEED_NOISE_STD_DEV            0.5
+#define NPS_GPS_SPEED_LATENCY                  0.2
+#define NPS_GPS_POS_NOISE_STD_DEV              2
+#define NPS_GPS_POS_BIAS_INITIAL_X             0e-1
+#define NPS_GPS_POS_BIAS_INITIAL_Y            -0e-1
+#define NPS_GPS_POS_BIAS_INITIAL_Z            -0e-1
+#define NPS_GPS_POS_BIAS_RANDOM_WALK_STD_DEV_X 1e-3
+#define NPS_GPS_POS_BIAS_RANDOM_WALK_STD_DEV_Y 1e-3
+#define NPS_GPS_POS_BIAS_RANDOM_WALK_STD_DEV_Z 1e-3
+#define NPS_GPS_POS_LATENCY                    0.2
+
+#endif /* GPS_PERFECT */
```

Comment:
Since these parameters are noise coefficients, maybe we should just put the first part in the default file and the second in the noisy file. Nobody is using the `GPS_PERFECT` flag option anyway.

Reply:
done